components/brave_ads/core/internal/processors/contextual/text_embedding/text_embedding_processor.cc

```diff
@@ -6,6 +6,7 @@
 #include "brave/components/brave_ads/core/internal/processors/contextual/text_embedding/text_embedding_processor.h"
 
 #include "base/check.h"
+#include "base/ranges/algorithm.h"
```

Comment:
Can we please add unit tests for this

Reply:
It does, but this is a follow up PR for that work.

Reply:
implementation is not trivial as processor adds results async to db https://github.com/brave/brave-browser/issues/29498

Reply:
Tests are sync, see other tests with database calls